feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validations/delegation/proxy/add/Declarations.kt

```diff
@@ -67,22 +70,18 @@ fun AddStakingProxyValidationSystemBuilder.maximumProxies(
     proxyRepository = proxyRepository
 )
 
-fun AddStakingProxyValidationSystemBuilder.enoughBalanceToPayDeposit(
-    proxyRepository: GetProxyRepository
-) = enoughBalanceToPayProxyDeposit(
-    chain = { it.chain },
-    accountId = { it.proxiedAccountId },
-    newDeposit = { it.depositWithQuantity.deposit },
-    availableBalance = { it.asset.freeInPlanks - it.asset.frozenInPlanks },
-    error = { payload, maxUsable ->
+fun AddStakingProxyValidationSystemBuilder.enoughBalanceToPayDepositAndFee() = sufficientBalance(
+    fee = { it.fee },
+    amount = { it.asset.token.configuration.amountFromPlanks(it.deltaDeposit) },
+    available = { it.asset.free - it.asset.frozen },
```

Comment:
I suggest to use methods from `Asset.Companion` to calculate transferable. In particular we can use `TransferableMode.REGULAR.calculateTransferable(free, frozen, reserved)`. That way we will keep transferable & ed calculation logic in a single place